src/main/java/org/spongepowered/common/mixin/core/entity/projectile/ThrowableEntityMixin.java

```diff
@@ -61,9 +61,9 @@
         ProjectileSourceSerializer.writeSourceToNbt(compound, ((Projectile) this).getShooter(), this.thrower);
     }
 
-    @Redirect(method = "onUpdate", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/projectile/EntityThrowable;onImpact(Lnet/minecraft/util/math/RayTraceResult;)V"))
+    @Redirect(method = "tick()V", at = @At(value = "INVOKE", target = "Lnet/minecraft/entity/projectile/ThrowableEntity;onImpact(Lnet/minecraft/util/math/RayTraceResult;)V"))
     public void onProjectileImpact(ThrowableEntity projectile, RayTraceResult movingObjectPosition) {
-        if (this.world.isRemote || movingObjectPosition.typeOfHit == RayTraceResult.Type.MISS) {
+        if (this.world.isRemote || movingObjectPosition.getType() == RayTraceResult.Type.MISS) {
```

Comment:
TODO will need to change to a `SpongeImplHooks.isMainThread` or `WorldBridge.bridge$isFake()` for the `world.isRemote.`